ios/Codeforces Watcher/Application/features/problems/SearchBarView.swift

```diff
@@ -0,0 +1,90 @@
+import SwiftUI
+
+struct SearchBarView: View {
+    
+    var onFilter: () -> Void = {}
+    
+    @State private var isShownSearchField = false
+    @State private var willShowSearchField = false
+    @State private var willHideSearchField = false
+    @State private var didShownSearchField = false
+    @State private var searchBoxText = ""
+    
+    private let textFieldAnimationDuration = 0.15
+    private let placehoderAnimationDelay = 0.2
+    
+    var body: some View {
+        HStack(spacing: 20) {
+            if !isShownSearchField {
```

Comment:
I'd like to split this long view into several local (private) components to improve readability a bit.